TestFoundation/TestCodable.swift

```diff
@@ -51,15 +51,17 @@ func expectRoundTripEquality<T : Codable>(of value: T, encode: (T) throws -> Dat
     XCTAssertEqual(value, decoded, "Decoded \(T.self) <\(decoded)> not equal to original <\(value)>")
 }
 
-func expectRoundTripEqualityThroughJSON<T : Codable>(for value: T) where T : Equatable {
+func expectRoundTripEqualityThroughJSON<T : Codable>(for value: T,
```

Comment:
That is what the [stdlib tests](https://github.com/apple/swift/blob/b8dc826d41a25d856d6e827eb67cbfef9c4f9705/test/stdlib/CodableTests.swift#L60) do though? I think we should follow the pattern from those. As far as I'm aware there's no extra cost involved in doing this.

Keeping the tests in sync will make it easier in future, if/when new tests are added we can port them over here more easily.